src/include/rbd/librbd.h

```diff
@@ -391,6 +391,18 @@ CEPH_RBD_API void rbd_mirror_image_status_list_cleanup(char **image_ids,
 CEPH_RBD_API int rbd_mirror_image_status_summary(rados_ioctx_t io_ctx,
     rbd_mirror_image_status_state_t *states, int *counts, size_t *maxlen);
 
+/* pool metadata */
+CEPH_RBD_API int rbd_pool_metadata_get(rados_ioctx_t io_ctx, const char *key,
+                                       char *value, size_t *val_len);
+CEPH_RBD_API int rbd_pool_metadata_set(rados_ioctx_t io_ctx, const char *key,
+                                       const char *value);
+CEPH_RBD_API int rbd_pool_metadata_remove(rados_ioctx_t io_ctx,
+                                          const char *key);
+CEPH_RBD_API int rbd_pool_metadata_list(rados_ioctx_t io_ctx, const char *start,
```

Comment:
@dillaman Thanks. I will add `rbd_config_list`. There are questions though:

Then I also assume we want `rbd_config_get/set/remove` functions that could access/modify a pool or image override?

Should  `rbd_config_list` return all possible ceph configuration variables (e.g. `debug ms`) or only `rbd_*`?

Also note that right although you can add any config variable to pool or image metadata, while only a small subset is actually applied (will override the config setting). I suppose it will confuse users. Do we want to improve this somehow? E.g. allow to set only these pool/image overrides that will be applied? And `rbd_config_list` could provide this info. 

After adding `rbd_config_get/set/remove/list` API will we still want `rbd_pool_metadata_get/set/remove/list` API similar to already existent image metadata API? It could be used to set any metadata for a pool, not only config overrides.